geopandas/tools/tests/test_clip.py

```diff
@@ -460,3 +468,13 @@ def test_clip_empty_mask(buffered_locations, mask):
     )
     clipped = clip(buffered_locations.geometry, mask)
     assert_geoseries_equal(clipped, GeoSeries([], crs="EPSG:3857"))
+
+
+def test_clip_sorting(disordered_point_gdf):
+    """Test the sorting kwarg in clip"""
+    bbox = shapely.geometry.box(0, 0, 2, 2)
+    unsorted_clipped_gdf = disordered_point_gdf.clip(bbox)
+    sorted_clipped_gdf = disordered_point_gdf.clip(bbox, sort=True)
+
+    assert not (sorted(unsorted_clipped_gdf.index) == unsorted_clipped_gdf.index).all()
+    assert (sorted(sorted_clipped_gdf.index) == sorted_clipped_gdf.index).all()
```

Comment:
Again, you can test against the explicit expected index.

Reply:
thanks, added